src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -166,37 +177,85 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
       exportAllOptions: [],
       exportAllSelectedOptions: {},
       isIncludeReferencesDeepChecked: true,
+      currentWorkspaceId: this.props.workspaces.currentWorkspaceId$.getValue(),
+      availableWorkspaces: this.props.workspaces.workspaceList$.getValue(),
+      workspaceEnabled: this.props.applications.capabilities.workspaces.enabled,
     };
   }
 
+  private get workspaceIdQuery() {
+    const { availableWorkspaces, currentWorkspaceId, workspaceEnabled } = this.state;
+    // workspace is turned off
+    if (!workspaceEnabled) {
+      return undefined;
+    } else {
+      // application home
+      if (!currentWorkspaceId) {
+        return availableWorkspaces?.map((ws) => ws.id);
+      } else {
+        return [currentWorkspaceId];
+      }
+    }
+  }
+
+  private get wsNameIdLookup() {
+    const { availableWorkspaces } = this.state;
+    //  Assumption: workspace name is unique across the system
+    return availableWorkspaces?.reduce((map, ws) => {
+      return map.set(ws.name, ws.id);
+    }, new Map<string, string>());
+  }
+
+  private formatWorkspaceIdParams<T extends { workspaces?: string[] }>(
+    obj: T
+  ): T | Omit<T, 'workspaces'> {
+    const { workspaces, ...others } = obj;
+    if (workspaces) {
+      return obj;
+    }
+    return others;
+  }
+
   componentDidMount() {
     this._isMounted = true;
+
+    this.fetchWorkspace();
     this.fetchSavedObjects();
     this.fetchCounts();
   }
 
   componentWillUnmount() {
     this._isMounted = false;
     this.debouncedFetchObjects.cancel();
+    this.currentWorkspaceIdSubscription?.unsubscribe();
+    this.workspacesSubscription?.unsubscribe();
   }
 
   fetchCounts = async () => {
     const { allowedTypes, namespaceRegistry } = this.props;
-    const { queryText, visibleTypes, visibleNamespaces } = parseQuery(this.state.activeQuery);
+    const { queryText, visibleTypes, visibleNamespaces, visibleWorkspaces } = parseQuery(
+      this.state.activeQuery
+    );
 
     const filteredTypes = filterQuery(allowedTypes, visibleTypes);
 
     const availableNamespaces = namespaceRegistry.getAll()?.map((ns) => ns.id) || [];
 
-    const filteredCountOptions: SavedObjectCountOptions = {
+    const filteredCountOptions: SavedObjectCountOptions = this.formatWorkspaceIdParams({
       typesToInclude: filteredTypes,
       searchString: queryText,
-    };
+      workspaces: this.workspaceIdQuery,
+    });
 
     if (availableNamespaces.length) {
       const filteredNamespaces = filterQuery(availableNamespaces, visibleNamespaces);
       filteredCountOptions.namespacesToInclude = filteredNamespaces;
     }
+    if (visibleWorkspaces?.length) {
+      filteredCountOptions.workspaces = visibleWorkspaces.map(
+        (wsName) => this.wsNameIdLookup?.get(wsName) || ''
```

Comment:
Do we need to filter out empty workspace id?

Reply:
sounds good, add filter logic